charts/vc-authn-oidc/templates/agent/deployment.yaml

```diff
@@ -0,0 +1,131 @@
+{{- $acapyFullName := include "acapy.fullname" . -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ $acapyFullName }}
+  labels:
+    {{- include "acapy.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.acapy.autoscaling.enabled }}
+  replicas: {{ .Values.acapy.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "acapy.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.acapy.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "acapy.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.acapy.image.pullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "acapy.serviceAccountName" . }}
+      automountServiceAccountToken: {{ .Values.acapy.serviceAccount.automountServiceAccountToken }}
+      securityContext:
+        {{- toYaml .Values.acapy.podSecurityContext | nindent 8 }}
+      containers:
+        - name: acapy
+          securityContext:
+            {{- toYaml .Values.acapy.containerSecurityContext | nindent 12 }}
+          image: "{{ .Values.acapy.image.repository }}:{{ .Values.acapy.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.acapy.image.pullPolicy }}
+          command:
+          - /bin/bash
+          args: [
+           "-c",
+           "aca-py start \
+           --inbound-transport http '0.0.0.0' {{ .Values.acapy.service.httpPort }} \
+           --outbound-transport http \
+           --admin '0.0.0.0' {{ .Values.acapy.service.adminPort }} \
+           --endpoint https://{{ include "vc-authn-oidc.host" . }} \
```

Comment:
We should likely use the `ACAPY_ENDPOINT` envvar/setting (see https://github.com/hyperledger/aries-cloudagent-python/blob/cbbcc8df2dbf1314e313adc170d3a0f77cbfed9c/aries_cloudagent/config/argparse.py#L606) and not force the protocol to `https`, but inherit what the route is using (unlikely to not use SSL, but having a hard-coded value here seems like it could potentially cause issues).

Reply:
Dynamically determining the protocol used in the endpoint is non-trivial. There are a number of caveats, is the end user using OpenShift Route object, Enables the Ingress object within the Values, or manually creates an Ingress object?

Perhaps we could add `agent.secureEnpoint` parameter in the Values, to specify wether to use `http` or `https`?

Reply:
If the consumer is creating endpoints manually, they will have to tweak things anyway so I am not too concerned about that. I would, however, want things to work out-of-the-box when deploying the chart as-is. My main gripe is that if they are NOT using TLS when deploying the solution (unlikely in OpenShift as it is provided by the cluster, more likely for oher k8s instances used for prototyping) there will be a scheme mismatch between `http` ingresses and `https` protocol required in the chart.

I am actually not sure the agent would work without SSL enabled on the routes, so maybe all we need to do is put proper instructions/warnings?